components/ArticleListContainer.js

```diff
@@ -1,11 +1,38 @@
 import styled from "styled-components";
 import ArticleCard from "./ArticleCard";
+import { useState } from "react";
 
-export default function ArticleListContainer({ articles }) {
+export default function ArticleListContainer({ currentArticles, delible }) {
+  const [articles, setArticles] = useState(currentArticles);
+  async function deleteArticle(articleId) {
+    try {
+      const response = await fetch(`/api/article/${articleId}`, {
+        method: "DELETE",
+      });
+      reloadArticle();
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  async function reloadArticle() {
+    try {
+      const response = await fetch(`/api/article`);
+      const favArticles = await response.json();
+      setArticles(favArticles);
```

Comment:
Why are you operating on a local state here? It seems redundant that you have an articles state in the store and this local state here. And here these two states get out of sync.

Reply:
As i mentioned above, the store (global variables) is for the articles results from the search. It s quite confusing because I also use ArticleListContainer and consequently ArtilceCard for displaying favourite articles from the database. So the local state here is to save favourite articles (in a collections) being fetched by server side rendering. Then if the article is deleted a reload is called and the local state will store the results from the reload. Otherwise it will just display the results from SSR

Reply:
oh wait a minute i think u re right. I dont need zustand for keeping the search results here coz I already pass down the results through the props. Sori hehehe